cmd/node/factory/structs.go

```diff
@@ -2114,6 +2118,7 @@ func newShardBlockProcessor(
 	statisticsProcessor process.ValidatorStatisticsProcessor,
 	bootStorer process.BootStorer,
 	gasSchedule map[string]map[string]uint64,
+	stateCheckpointModulus uint,
```

Comment:
for now it can go like this, but this will have to be changed to generate the checkpoint on a trigger.

Reply:
Agreed. Added TODO.